docker-compose.yml

```diff
@@ -2,16 +2,45 @@ version: '3.8'
 services:
   mysql:
     image: mysql:8.0
-    command: --default-authentication-plugin=mysql_native_password
+    restart: always
+    command:
+      --default-authentication-plugin=mysql_native_password
+      --bind-address=0.0.0.0
     cap_add:
       - SYS_NICE
     ports:
-      - '3306:3306'
+      - 3306:3306
     environment:
       - MYSQL_ROOT_PASSWORD=default_password
       - MYSQL_DATABASE=checkpoint
     volumes:
       - mysql:/var/lib/mysql
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+    healthcheck:
+      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost", "-uroot", "-ppass"]
+      interval: 5s
+      timeout: 5s
+      retries: 20
+    network_mode: bridge
+
+  node:
+    build: .
+    command: yarn dev
+    restart: always
+    ports:
+      - 3000:3000
+    depends_on:
+      mysql:
+        condition: service_healthy
+    environment:
+      - DATABASE_URL=mysql://root:default_password@mysql:3306/checkpoint
```

Comment:
I recommend splitting the DATABASE_URL and DB_NAME. It gives more flexibility and you will able to reuse DB_NAME for `node` and `mysql` services

Reply:
I prefer if we dont do this change we use connection string everywhere, this make setup more complicated

Reply:
agree. Joining substrings is much easier than single out them. But we should have single source of truth